### PR TITLE
Update otel collector install script urls

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
## Purpose of this pull request

We're in the process of moving all of our packaging assets from the main [otel collector repository](https://github.com/SumoLogic/sumologic-otel-collector) to a dedicated [packaging repository](https://github.com/SumoLogic/sumologic-otel-collector-packaging). This move is complete for the install scripts, and I'd like to update the urls in the documentation.

Some of the links were also incorrect or outdated, so I corrected them.

## Select the type of change

- [x] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions, updating sections
- [ ] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - .clabot, version updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

### CODE REVIEW

**Review by sumo-drosiek — APPROVED** [review]

File: docs/send-data/opentelemetry-collector/install-collector/linux.md [hunk 1]
```diff
@@ -60,11 +60,11 @@ You can run the script in two ways:
 
 * By piping `curl` straight into `bash`:
    ```bash
-   curl -Ls https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh | sudo -E bash -s -- --tag "host.group=default" --tag "deployment.environment=default"
+   curl -Ls https://github.com/SumoLogic/sumologic-otel-collector-packaging/releases/latest/download/install.sh | sudo -E bash -s -- --tag "host.group=default" --tag "deployment.environment=default"
    ```
 * By first downloading the script, inspecting its contents for security, and then running it:
    ```bash
-   curl -Lso install-otelcol-sumo.sh https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh
+   curl -Lso install-otelcol-sumo.sh https://github.com/SumoLogic/sumologic-otel-collector-packaging/releases/latest/download/install.sh
```
Comment (rnishtala-sumo) [inline docs/send-data/opentelemetry-collector/install-collector/linux.md:67]: should we also add a link here for a customer that tries to install an older 0.95.x release?

Comment (sumo-drosiek) [reply docs/send-data/opentelemetry-collector/install-collector/linux.md:67]: The latest install script should support `v0.95.x`. Am I correct?

Comment (swiatekm-sumo) [reply docs/send-data/opentelemetry-collector/install-collector/linux.md:67]: They can do that using the latest install script by using the `--version` command line flag. The install script is highly backwards compatible, at least until around 0.76.0.

**Review by rnishtala-sumo — APPROVED** [review]

**Review by JV0812 — APPROVED** [review]
LGTM

**Comment by kimsauce** [comment]
thanks @swiatekm-sumo for doing this!

**Pull request merged**